src/Cake.Core/Scripting/Processors/LoadDirectiveProcessor.cs

```diff
@@ -46,10 +48,44 @@ public override bool Process(IScriptProcessor processor, ScriptProcessorContext
                 return false;
             }
 
+            var scriptPathString = tokens[1].UnQuote();
+            if (string.IsNullOrWhiteSpace(scriptPathString))
+            {
+                throw new CakeException(
+                    string.Format(CultureInfo.InvariantCulture,
+                        "{0}{2}{1}",
+                        "Load directive may not contain empty path or Uri",
+                        line,
+                        Environment.NewLine));
+            }
+
             var directoryPath = GetAbsoluteDirectory(currentScriptPath);
-            var scriptPath = new FilePath(tokens[1].UnQuote()).MakeAbsolute(directoryPath);
+            ScriptReference scriptReference;
+            Uri scriptUri;
+            if (!Uri.TryCreate(scriptPathString, UriKind.RelativeOrAbsolute, out scriptUri)
+                || !scriptUri.IsAbsoluteUri || scriptUri.IsFile)
+            {
+                var scriptPath = new FilePath(scriptPathString);
+                scriptReference = new ScriptReference(scriptPath);
+            }
+            else
+            {
+                // Download the script to a local directory.
+                FilePath tempFile = System.IO.Path.GetTempFileName();
+                try
+                {
+                    var client = new WebClient();
```

Comment:
Should make this mockable so we can write tests for this. Maybe a `IWebClient` in `Cake.Core.Net`?